lib/internal/modules/esm/assets/load.js

```diff
@@ -0,0 +1,300 @@
+'use strict';
+const {
+  globalThis,
+  ArrayFrom,
+  ArrayPrototypeJoin,
+  ArrayPrototypePush,
+  ArrayPrototypeSplice,
+  Promise,
+  PromisePrototypeThen,
+  PromiseResolve,
+  SafeMap,
+  StringPrototypeEndsWith,
+  StringPrototypePadStart,
+  StringPrototypeSlice,
+  StringPrototypeSplit,
+  StringPrototypeStartsWith,
+} = primordials;
+const {
+  Buffer: {
+    concat: BufferConcat
+  }
+} = globalThis;
+const {
+  codes: {
+    ERR_UNSUPPORTED_ESM_NETWORK_IMPORT_SPECIFIER,
+    ERR_UNSUPPORTED_ESM_NETWORK_BAD_RESPONSE
+  }
+} = require('internal/errors');
+const { URL } = require('internal/url');
+
+
+/**
+ * Lazily load these to avoid snapshot issues
+ */
+let HTTPSAgent;
+function HTTPSGet(url, opts) {
+  const https = require('https');
+  HTTPSAgent = HTTPSAgent || new https.Agent({
+    keepAlive: true
+  });
+  return https.get(url, {
+    agent: HTTPSAgent,
+    ...opts
+  });
+}
+
+let HTTPAgent;
+function HTTPGet(url, opts) {
+  const http = require('http');
+  HTTPAgent = HTTPAgent || new http.Agent({
+    keepAlive: true
+  });
+  return http.get(url, {
+    agent: HTTPAgent,
+    ...opts
+  });
+}
+
+function dnsLookup(name, opts) {
+  // eslint-disable-next-line no-func-assign
+  dnsLookup = require('dns/promises').lookup;
+  return dnsLookup(name, opts);
+}
+
+function createBrotliDecompress() {
+  // eslint-disable-next-line no-func-assign
+  createBrotliDecompress = require('zlib').createBrotliDecompress;
+  return createBrotliDecompress();
+}
+
+function createUnzip() {
+  // eslint-disable-next-line no-func-assign
+  createUnzip = require('zlib').createUnzip;
+  return createUnzip();
+}
+
+/**
+ * @param {URL} parsed
+ * @returns {Promise<CacheEntry> | CacheEntry}
+ */
+function fetchWithRedirects(parsed) {
+  const existing = getCache.get(parsed.href);
+  if (existing) {
+    return existing;
+  }
+  const handler = parsed.protocol === 'http:' ? HTTPGet : HTTPSGet;
+  const result = new Promise((fulfill, reject) => {
+    const req = handler(parsed, {
+      headers: {
+        Accept: '*/*'
+      }
+    })
+    .on('error', reject)
+    .on('response', (res) => {
+      function dispose() {
+        req.destroy();
+        res.destroy();
+      }
+      if (res.statusCode >= 300 && res.statusCode <= 303) {
```

Comment:
use primordials, but sure, this will be slightly slower / require reading the list rather than being readable in the if itself which would be unfortunate

Reply:
Maybe it's just me—I found parsing the conditions less grokkable (whereas reading the list to be easy, and the description from the var name easier still). I think the perf hit is trivial, no?

It's just a nit though.

Reply:
There’s also:

```suggestion
      if (/300|301|302|303/.test(res.statusCode)) {
```

Or:

```suggestion
      if (/30[0|1|2|3]/.test(res.statusCode)) {
```

Reply:
I would prefer the first—I just found reading the numbers quicker than "greater than or equal to 300 and less than or equal to 303".

Reply:
yes, regex `test` is another thing primordials are needed for.